src/controller/InputValidator.js

```diff
@@ -0,0 +1,42 @@
+import {
+  CAR_NAME_EMPTY,
+  CAR_NAME_LIMIT_LENGTH,
+  VALID_NAME_PATTERN,
+} from '../constants/constraints.js';
+import {
+  ERROR_ATTEMPT_COUNT_MUST_BE_POSITIVE,
+  ERROR_CAR_NAME_LENGTH,
+  ERROR_DUPLCIATE_CAR_NAMES,
+  ERROR_EMPTY_CAR_NAME,
+  ERROR_INVALID_INPUT_FORMAT,
+} from '../constants/errorMessage.js';
+import throwError from '../Error/handleError.js';
+
+class InputValidator {
+  static validateCarName(names) {
+    if (!VALID_NAME_PATTERN.test(names)) {
+      throwError(ERROR_INVALID_INPUT_FORMAT);
+    }
+    if (names.some((name) => name.length === CAR_NAME_EMPTY)) {
+      throwError(ERROR_EMPTY_CAR_NAME);
+    }
+    if (names.some((name) => name.length > CAR_NAME_LIMIT_LENGTH)) {
+      throwError(ERROR_CAR_NAME_LENGTH);
+    }
+    const hasDuplicates = names.length !== new Set(names).size;
+    if (hasDuplicates) {
+      throwError(ERROR_DUPLCIATE_CAR_NAMES);
+    }
+  }
```

Comment:
validateCarName은 carName의 유효성을 검증하는 함수로 보여요.

하지만 각각의 유효성 검증을 이해하기 위해서는 조건문이 하는 역할을 먼저 이해해야만해요. 

이 부분을 따로 함수로 작성해보면 어떨까요? 

상세히 조건문을 해석하지 않아도 어떤 유효성을 검증하는지 이해할 수 있어요!

```suggestion
  static validateCarName(names) {
    InputValidator.validateInvalidInput(names);
    InputValidator.validateEmptyCarName(names);
    InputValidator.validateCarNameLength(names);
    InputValidator.validateHasDuplicated(names);
  }
```

Reply:
훨씬 깔끔하고 가독성도 좋은 것 같아요 감사합니다!